src/app.ts

```diff
@@ -777,6 +788,43 @@ export class App {
 
     ensureDirectoryExists(destinationDirectory);
 
+    if (
+      newRevisionsOnly &&
+      fs.existsSync(path.join(destinationDirectory, 'index.d.ts'))
+    ) {
+      let existingRevision;
+      await eachLine(
+        path.join(destinationDirectory, 'index.d.ts'),
+        {},
+        line => {
+          if (line.startsWith('// Revision: ')) {
+            const match = line.match(/^\/\/ Revision\: (\d+)$/);
+            if (match !== null && match.length === 2) {
+              existingRevision = Number(match[1]);
+              return false;
+            }
+          }
+          return true;
+        }
+      );
+
+      if (!api.revision) {
+        return console.error(`There's no revision in JSON: ${api.id}`);
+      }
+      if (!existingRevision) {
+        return console.error(
+          `Can't find previous revision in index.d.ts: ${api.id}`
+        );
+      }
+
+      const newRevision = Number(api.revision);
+      if (existingRevision > newRevision) {
+        return console.warn(
+          `Local revision ${existingRevision} is more recent than fetched ${newRevision}, skipping ${api.id}`
+        );
+      }
```

Comment:
> I added solution for #81 in this PR. I think it could be refactored, but it works quiet well, take a look, please

Yeah, this seems fine!

It would be nice not to keep bloating `app.ts`, but this seems to be the best spot for this at the moment. I might start giving some thought into how we could move software architecture in a better direction and ultimately get some more test coverage.

Reply:
Yes, I'm also considering rewriting the core generator code from the ground-up. Too much magic is going on there :)
So yes, any refactoring and architecture discussions are very welcome!

Reply:
Cool, I would personally have to give this some more thought, even before summarizing anything in an issue, but I will certainly create one when I get my head around it!